nrich-notification/README.md

```diff
@@ -33,33 +46,76 @@ To be able to use this module following configuration is required:
+### [ConstraintConversionService][constraint-conversion-service-url]
+
+[`ConstraintConversionService`][constraint-conversion-service-url] is a service responsible for converting `javax.validation.ConstraintViolation` list to Spring `Errors` instance.
+This conversion is done to make the implementation of [`NotificationResolverService`][notification-resolver-service-url] less complex since it only needs to work with Spring's `Errors` when processing
+validation errors.
+
+---
 
-`NotificationMessageResolverService` is a service responsible for resolving messages from message codes and
-`ObjectErrors`. Default implementation is `MessageSourceNotificationMessageResolverService` that resolves messages from Spring's `MessageSource`.
+### [NotificationMessageResolverService][nrich-notification-message-resolver-service-url]
 
-`NotificationResolverService` is main service that is responsible for creating notifications. It can create notifications for validation errors, exceptions or action names.
-Consequently, it accepts either Spring's `Errors` or Jakarta `ConstraintViolationException` for creating validation notifications, `Throwable` for creating error notifications and
-`String` for action notification.
+[`NotificationMessageResolverService`][nrich-notification-message-resolver-service-url] is a service responsible for resolving messages from message codes and `ObjectErrors`.
+Default implementation is [`MessageSourceNotificationMessageResolverService`][message-source-notification-message-resolver-service-url] that resolves messages from Spring's `MessageSource`.
 
-`NotificationResponseService` is a service that adds support for returning data with notifications by introducing additional methods that return `NotificationDataResponse`.
+---
 
-When using Spring Web MVC users can use `NotificationResponseService` implementation `WebMvcNotificationResponseService` that is a wrapper around `NotificationResolverService` providing a
-convenient way of returning a notification with a response.
+### [NotificationResolverService][notification-resolver-service-url]
+
+[`NotificationResolverService`][notification-resolver-service-url] is the core service that is responsible for creating notifications.
+It can create notifications for validation errors, exceptions or action names.
+Consequently, it accepts either Spring's `Errors` or Jakarta `ConstraintViolationException` for creating validation notifications, `Throwable` for creating error notifications and `String` for action
+notification.
+
+---
+
+### [NotificationResponseService][notification-response-service-url]
+
+[`NotificationResponseService`][notification-response-service-url] is a service that adds support for additional data in notifications by introducing additional methods that
+return [`NotificationDataResponse`][notification-data-response-url].
+When using Spring Web MVC users can use [`NotificationResponseService`][notification-response-service-url] implementation
```

Comment:
"When using Spring Web MVC users can use `WebMvcNotificationResponseService` implementation that is..." seems clear enough to me in a given context.

Reply:
Agreed